index.js

```diff
@@ -429,6 +435,141 @@ class ClientServiceBase {
   _isNodeJsRuntime() {
     return typeof window === 'undefined';
   }
+
+  /**
+   * @return {Promise<CertificateRegistrationRequest>}
+   */
+  async _createCertificateRegistrationRequest() {
+    const builder = new CertificateRegistrationRequestBuilder(
+        new this.protobuf.CertificateRegistrationRequest(),
+    ).withCertHolderId(this.certHolderId)
+        .withCertVersion(this.certVersion)
+        .withCertPem(this.certPem);
+
+    return builder.build();
+  }
+
+  /**
+   * @param {string} id of the function
+   * @param {string} name of the function
+   * @param {Uint8Array} functionBytes of the function
+   * @return {Promise<FunctionRegistrationRequest>}
+   * @throws {ClientError}
+   */
+  async _createFunctionRegistrationRequest(id, name, functionBytes) {
+    if (!(functionBytes instanceof Uint8Array)) {
+      throw new ClientError(
+          StatusCode.CLIENT_IO_ERROR,
+          'parameter functionBytes is not a \'Uint8Array\'',
+      );
+    }
+
+    const builder = new FunctionRegistrationRequestBuilder(
+        new this.protobuf.FunctionRegistrationRequest(),
+    ).withFunctionId(id)
+        .withFunctionBinaryName(name)
+        .withFunctionByteCode(functionBytes);
+
+    return builder.build();
+  }
+
+  /**
+   * @param {string} id of the contract
+   * @param {string} name  the canonical name of the contract class.
+   *  For example "com.banking.contract1"
+   * @param {Uint8Array} contractBytes
+   * @param {Object}  [properties]
+   *  JSON Object used for setting client properties
+   * @return {Promise<ContractRegistrationRequest>}
+   * @throws {ClientError}
+   */
+  async _createContractRegistrationRequest(
+      id, name, contractBytes, properties,
+  ) {
+    if (!(contractBytes instanceof Uint8Array)) {
+      throw new ClientError(
+          StatusCode.CLIENT_IO_ERROR,
+          'parameter contractBytes is not a \'Uint8Array\'',
+      );
+    }
+
+    const propertiesJson = JSON.stringify(properties);
+    const builder = new ContractRegistrationRequestBuilder(
+        new this.protobuf.ContractRegistrationRequest(),
+        this.signer,
+    ).withContractId(id)
+        .withContractBinaryName(name)
+        .withContractByteCode(contractBytes)
+        .withContractProperties(propertiesJson)
+        .withCertHolderId(this.certHolderId)
+        .withCertVersion(this.certVersion);
+
+    try {
+      return builder.build();
+    } catch (e) {
+      throw new ClientError(
+          StatusCode.RUNTIME_ERROR,
+          e.message,
+      );
+    }
+  }
+
+  /**
+   * Validate the integrity of an asset
```

Comment:
I think we should have this message in the caller function.

Reply:
I am sorry. It is a wrong copy/paste.
I will fix it.